packages/vite/src/node/optimizer/optimizer.ts

```diff
@@ -408,6 +456,16 @@ async function createDepsOptimizer(
           newDepsToLogHandle = setTimeout(() => {
             newDepsToLogHandle = undefined
             logNewlyDiscoveredDeps()
+            if (warnAboutMissedDependencies) {
+              logDiscoveredDepsWhileScanning()
+              config.logger.info(
```

Comment:
This and the length check in `logDiscoveredDepsWhileScanning` feels like there is some duplication of state maybe?
Could it be possible to have the magenta log without the missed log one?
Maybe these checks and this two logs could be grouped (just passing a `hint: string` as an argument)

I have still to understand the difference between the two list of deps so maybe this will make sens eafter

Reply:
in case it helps https://github.com/vitejs/vite/pull/15244/files#r1417407827